source/blog/042-wide-gamut-colors-in-sass.md

```diff
@@ -0,0 +1,279 @@
+
+In general, any color defined in a given space will remain in that space, and be emitted in that space. The space is defined by the function used, either one of the named spaced passed to `color()`, or the function name (e.g. `lab` for colors defined using the `lab()` function).
+
+However, the `rgb`, `hsl`, and `hwb` spaces will be considered "legacy spaces", and will often get special handling for the sake of backwards compatibility. Legacy colors will continue to be emitted in the most backwards-compatible format available. This matches CSS’s own backwards-compatibility behavior. Colors defined using hex notation or CSS color names are also considered part of the legacy `rgb` color space.
+
+Sass provides a variety of tools for inspecting and working with these color spaces:
+
+- We can inspect the space of a color using `color.space($color)`
+- We can ask if the color is in a legacy space with `color.is-legacy($color)`
+- We can *convert* a color from one space to another using `color.to-space($color, $space)`
+
+All of these functions are provided by the built-in [Sass Color Module](https://sass-lang.com/documentation/modules/color/):
+
+```scss
+@use 'sass:color';
+$brand: MediumVioletRed;
+
+// results: rgb, true
+$initial: color.space($brand);
+$is-legacy: color.is-legacy($brand);
+
+// result: oklch(55.34% 0.2217 349.7)
+$converted: color.to-space($brand, 'oklch');
+
+// results: oklch, false
+$new-space: color.space($brand);
+$is-legacy: color.is-legacy($brand);
+```
+
+Once we can convert a color between spaces, we no longer consider those colors to be *equal*. But we can ask if they would render as 'the same' color, using the `color.same()` function:
+
+```scss
+@use 'sass:color';
+$orange-rgb: #ff5f00;
+$orange-oklch: oklch(68.72% 20.966858279% 41.4189852913deg);
+
+// result: false
+$equal: $orange-rgb == $orange-oklch;
+
+// result: true
+$same: color.same($orange-rgb, $orange-oklch);
+```
+
+We can inspect the individual channels of a color using `color.channel()`. By default, it only supports channels that are available in the color's own space, but we can pass the `$space` parameter to return the value of the channel value after converting to the given space:
+
+```scss
+@use 'sass:color';
+$brand: hsl(0 100% 25.1%);
+
+// result: 25.1%
+$hsl-lightness: color.channel($brand, "lightness");
+
+// result: 37.67%
+$oklch-lightness: color.channel($brand, "lightness", $space: oklch);
+```
+
+CSS has also introduced the concept of 'powerless' and 'missing' color channels. For example, an `hsl` color with `100%` lightness will *always be white*. In that case, we can consider both the `hue` and `saturation` channels to be powerless. Changing their value won't have any impact on the resulting color. Sass allows us to ask if a channel is powerless using the `color.is-powerless()` function:
```

Comment:
Actually, according to the current spec, only `0%` saturation causes the hue to become powerless. Lightness _never_ causes powerlessness. (This is actually wrong in our implementation too :cold_sweat:.)